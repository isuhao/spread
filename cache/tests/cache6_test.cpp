#include <iostream>

#include "index.hpp"

using namespace Spread;
using namespace std;

Cache::CacheIndex index;
Hash hello("hello",5);

void dump()
{
  cout << "INDEX:\n";
  Cache::CIVector vec;
  index.getEntries(vec);
  for(int i=0; i<vec.size(); i++)
    {
      Cache::CIEntry &e = vec[i];
      cout << e.hash << " " << e.file << " " << e.writeTime << endl;
    }
  cout << endl;
}

int main()
{
  Hash::DirMap files;
  files["hello.dat"] = hello;
  files["test.sh"];
  files["nothing"];

  cout << "Checking files:\n";
  index.checkMany(files);
  cout << "  test.sh: " << files["test.sh"] << endl;
  cout << "  nothing: " << files["nothing"] << endl;
  dump();

  cout << "Updating list:\n";
  files.clear();
  files["hello.dat"];
  files["hello2.dat"];
  Cache::StrSet st;
  st.insert("test.sh");
  index.addMany(files, st);
  dump();

  return 0;
}
