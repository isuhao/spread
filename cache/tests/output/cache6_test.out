Checking files:
  test.sh: ctEjJBRstghw4_UpmjBdhwJZFl8faISyIeEk2sOH5LLfAQ
  nothing: 00
INDEX:
LPJNul-wow4m6DsqxbninhsWHlwfp0JecwQzYpOLmCQF /home/mortennk/koding/spread/cache/tests/hello.dat 1341231752
ctEjJBRstghw4_UpmjBdhwJZFl8faISyIeEk2sOH5LLfAQ /home/mortennk/koding/spread/cache/tests/test.sh 1341224553

Updating list:
INDEX:
LPJNul-wow4m6DsqxbninhsWHlwfp0JecwQzYpOLmCQF /home/mortennk/koding/spread/cache/tests/hello.dat 1341231752
LPJNul-wow4m6DsqxbninhsWHlwfp0JecwQzYpOLmCQF /home/mortennk/koding/spread/cache/tests/hello2.dat 1341232094

